.github/workflows/rust-build.yml

```
@@ -19,85 +19,62 @@ on:
         description: 'Publish package to crates.io'
         default: false
         type: boolean
-      upload-artifact:
-        description: 'Upload build artifact'
-        default: false
-        type: boolean
-      artifact-name:
-        description: 'Name of the artifact to upload'
-        type: string
-        required: false
-      artifact-path:
-        description: 'Path to the artifact to upload'
-        type: string
-        required: false
     secrets:
-      CRATES_IO_TOKEN:
+      CARGO_REGISTRY_TOKEN:
         required: false
 
+env:
+  CARGO_TERM_COLOR: always
+  CARGO_REGISTRY_TOKEN: ${{ secrets.CARGO_REGISTRY_TOKEN }}
+  
 jobs:
-  build:
+  build_and_publish:
     runs-on: ubuntu-latest
-    outputs:
-      build_success: ${{ steps.set-output.outputs.build_success }}
     steps:
       - name: Checkout repository
         uses: actions/checkout@v4
 
       - name: Install Rust toolchain
-        uses: actions-rs/toolchain@v1
+        uses: dtolnay/rust-toolchain@stable
```

To verify if the spelling is correct, but this is the documented way to select the version to install (the toolchain parameter has to be removed)
See https://github.com/dtolnay/rust-toolchain?tab=readme-ov-file#example-workflow
```suggestion
        uses: dtolnay/rust-toolchain@${{ inputs.rust-version }}
```